### PR TITLE
PHOENIX-7383 Unify in-memory representation of Conditional TTL expressions and Literal TTL values

### PR DESCRIPTION
**Pull request by tkhurana**
This PR lays the framework of Conditional TTL.

1. Introduces a new abstract class to hold both conditional TTL and literal TTL values in PTable
2. Simplifies the TTLTracker interface in PhoenixCompactionScanner
3. Introduces the hooks in PhoenixCompactionScanner to work with any type of TTL expression
4. Basic DDL checks in Create/Alter table for Conditional TTL

Tests ran locally:
ViewTTLIT, TTLAsPhoenixTTLIT, TableTTLIT

### CODE REVIEW

**Review by jpisaac — COMMENTED** [review]
Took a first pass. Looks good, will take another look today.

File: phoenix-core-client/src/main/java/org/apache/phoenix/schema/ConditionTTLExpression.java [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+        // Conditional TTL is not sent as a scan attribute
+        // Masking is implemented using query re-write
+        return null;
+    }
+
+    /**
+     * Validates that all the columns used in the conditional TTL expression are present in the table
+     * or its parent table in case of view
+     */
+    private static class VerifyCreateConditionalTTLExpression extends ExpressionCompiler {
```
Comment (jpisaac) [inline phoenix-core-client/src/main/java/org/apache/phoenix/schema/ConditionTTLExpression.java:132]: Can we restrict the expression to not include aggregate functions?

File: phoenix-core/src/test/java/org/apache/phoenix/schema/ConditionalTTLExpressionDDLTest.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        String ddlTemplate = "create table %s (k1 bigint not null, k2 bigint not null, col1 varchar " +
+                "constraint pk primary key (k1,k2 desc)) TTL = '%s'";
+        String ttl = "PHOENIX_ROW_TIMESTAMP() < CURRENT_DATE() - 100";
+        String tableName = generateUniqueName();
+        String ddl = String.format(ddlTemplate, tableName, ttl);
+        try (Connection conn = DriverManager.getConnection(getUrl(), PropertiesUtil.deepCopy(TEST_PROPERTIES))) {
+            conn.createStatement().execute(ddl);
+            assertConditonTTL(conn, tableName, ttl);
+        }
+    }
```
Comment (jpisaac) [inline phoenix-core/src/test/java/org/apache/phoenix/schema/ConditionalTTLExpressionDDLTest.java:128]: Should add tests for views and indexes.

**Review by jpisaac — COMMENTED** [review]
LGTM, just some  comments on more validation and tests

**Comment by tkhurana** [comment]
@jpisaac Added check for aggregate function and some more unit tests for views. There are more IT tests for views and indexes in TTLAsPhoenixTTLIT which have been enhanced to work with Conditional TTL

**Review by jpisaac — APPROVED** [review]

**Pull request merged**